### PR TITLE
Add accountID to GC message

### PR DESCRIPTION
**Pull request by KillianG**
Issue: BB-503

### CODE REVIEW

**Review by welansari — APPROVED** [review]

File: extensions/lifecycle/tasks/LifecycleUpdateExpirationTask.js [hunk 1]
```diff
@@ -126,7 +126,8 @@ class LifecycleUpdateExpirationTask extends BackbeatTask {
               })
               .setAttribute('source', entry.getAttribute('source'))
               .setAttribute('serviceName', 'lifecycle-transition')
-              .setAttribute('target.locations', locations);
+              .setAttribute('target.locations', locations)
+              .setAttribute('target.accountId', accountId);
```
Comment (francoisferrand) [inline extensions/lifecycle/tasks/LifecycleUpdateExpirationTask.js]: should also (try to) pass the `ownerId`, for good measure (like in the other cases)

File: tests/unit/gc/GarbageCollectorTask.spec.js [hunk 2]
```diff
@@ -199,4 +199,33 @@ describe('GarbageCollectorTask', () => {
...
+                  versionId: version,
+              })
+              .setAttribute('serviceName', 'lifecycle-transition')
+              .setAttribute('target.oldLocation', 'old-location')
+              .setAttribute('target.newLocation', 'new-location')
+              .setAttribute('target.bucket', bucket)
+              .setAttribute('target.key', version)
+              .setAttribute('target.version', key)
+              .setAttribute('target.accountId', accountId)
+              .setAttribute('target.owner', owner);
```
Comment (francoisferrand) [inline tests/unit/gc/GarbageCollectorTask.spec.js:221]: should it not be `target.ownerId` ?

File: extensions/lifecycle/tasks/LifecycleColdStatusArchiveTask.js [hunk 3]
```diff
@@ -36,7 +37,7 @@ class LifecycleColdStatusArchiveTask extends LifecycleUpdateTransitionTask {
               .setAttribute('target.key', key)
               .setAttribute('target.version', version)
               .setAttribute('target.accountId', accountId)
-              .setAttribute('target.owner', owner);
+              .setAttribute('target.ownerId', ownerId);
```
Comment (francoisferrand) [inline extensions/lifecycle/tasks/LifecycleColdStatusArchiveTask.js]: this is a partial fix only.

Looking at backbeat (lifecycle) overall, there are _lots_ places where we set `target.owner` field : so either we should change them all, or change GC to extract the `owner` field from message.

File: extensions/gc/tasks/GarbageCollectorTask.js [hunk 4]
```diff
@@ -23,16 +23,16 @@ class GarbageCollectorTask extends BackbeatTask {
     // account id information is missing.
     // TODO BB-367: replace once replication services uses assume role
     _getAccountId(entry, log, cb) {
-        const { accountId, ownerId } = entry.getAttribute('target');
+        const { accountId, owner } = entry.getAttribute('target');
```
Comment (francoisferrand) [inline extensions/gc/tasks/GarbageCollectorTask.js:26]: also needs to be changed in https://github.com/scality/backbeat/blob/c790d6de0e52b1ba701dd1fe7103362fb97ed093/extensions/replication/tasks/UpdateReplicationStatus.js#L279

**Review by francoisferrand — APPROVED** [review]

**Pull request merged**